### ISSUE TITLE
Troubles with rendering on Samsung Galaxy Nexus GT-I9250

### ISSUE THREAD

**Issue by klblk** [mapbox/mapbox-gl-native#8735]
**Platform:** Android (Samsung Galaxy Nexus GT-I9250 - PowerVR SGX540 - Android 4.3)
**Mapbox SDK version:** 5.0.2

### Steps to trigger behavior

 Mapbox demo app -> Styles -> A simple map view

### Actual behavior
![photo_2017-04-13_09-02-49](https://cloud.githubusercontent.com/assets/6357562/24987778/f5ec283e-202b-11e7-9299-15eb2117b96b.jpg)
![photo_2017-04-13_09-01-16](https://cloud.githubusercontent.com/assets/6357562/24987782/fbbdc83a-202b-11e7-9a66-fc63a0b730f6.jpg)
![photo_2017-04-13_08-59-26](https://cloud.githubusercontent.com/assets/6357562/24987791/ff2f0330-202b-11e7-953b-7a67a47738af.jpg)

**Comment by jfirebaugh** [comment]
Though it looks different than the screenshots attached to #8658, I did see rendering errors somewhat like this while fixing that issue. The fix for it is in 5.1.1, released today -- could you check whether this is still reproducible in that version?

**Comment by ivovandongen** [comment]
@jfirebaugh Tried to verify this (have the same galaxy nexus here), but the test app crashes on 5.1.1 due to:

```
07-24 15:24:34.060 3736-3736/com.mapbox.mapboxsdk.testapp E/AndroidRuntime: FATAL EXCEPTION: main
                                                                            java.lang.Error: glBindAttribLocation(id, location, name): Error GL_INVALID_VALUE at /Users/ivo/git/mapbox-gl-native/src/mbgl/gl/attribute.cpp:9
```

This looks very much like https://github.com/mapbox/mapbox-gl-native/issues/9331 which got fixed in https://github.com/mapbox/mapbox-gl-native/pull/9373 but reverted on the release branch in https://github.com/mapbox/mapbox-gl-native/commit/441644166310314b5e8016405691110e69a0a3cd because it caused other issues it seems.

**Comment by jfirebaugh** [comment]
Release branch doesn't include #9250 though, so I wouldn't expect it to have gone over the 8 attribute limit.

**Comment by ivovandongen** [comment]
@jfirebaugh Sorry, you're right. I just switched branches again to debug this and it works like a charm. Must have gotten something crossed yesterday.

Also, now I can confirm that this fixes the issue in the OP:

![screenshot_2017-07-25-11-45-04](https://user-images.githubusercontent.com/64527/28563573-08dfe148-712f-11e7-967f-e73ee492922b.png)

**Comment by tobrun** [comment]
Adding context here that this fix shipped with v5.1.1 :tada: Closing.

**Comment by jfirebaugh** [comment]
Hooray!

BTW the text rendering issue seen in the above screenshot is #6922, fixed for 5.2.0.
